classes/generate_people.rb

```diff
@@ -0,0 +1,70 @@
+require './student'
+require './teacher'
+require './data_values'
+require './classroom'
+
+class GeneratePeople
+  include DataValues
+
+  def initialize
+    @people = DataValues.people
+    @students = DataValues.students
+    @teachers = DataValues.teachers
+  end
+
+  def create_person
+    puts 'Which do you want to create? A student (press 1) or a teacher (press 2)?'
+    num = gets.chomp.to_i
+    case num
+    when 1
+      student_option
+    when 2
+      teacher_option
+    end
+  end
+
+  def student_option
+    puts "Enter Student's name:"
+    name = gets.chomp
+    puts "Enter Student's age:"
+    age = gets.chomp.to_i
+    puts "Enter Student's classroom Name:"
+    croom = gets.chomp
+    classroom = Classroom.new(croom)
+    parent_permission = permission?
```

Comment:
- [ ] Please separate the input fields for name, age and classroom in a different method to make `student_option`  cleaner. It can be something similar to this
```
  def student_option
    age, name = age_and_name

    croom = classroom

    perm = permission

    student = Student.new(age, croom, name: name, parent_permission: perm == 'Y')
    @people << student
  end

  def age_and_name
    print 'Enter students age:'
    age = gets.chomp.to_i
    print 'Enter Students name: '
    name = gets.chomp
    [age, name]
  end

  def classroom
    print 'Classroom:'
    gets.chomp
  end
```

Reply:
Thanks, @Lucas-Masaba for this great lesson.

Reply:
Awesome @raminka13